src/jtag3.c

```diff
@@ -2499,7 +2542,7 @@ int jtag3_read_sib(const PROGRAMMER *pgm, const AVRPART *p, char *sib) {
     return status;
 
   memcpy(sib, resp+3, AVR_SIBLEN);
-  sib[AVR_SIBLEN] = 0; // Zero terminate string
+  sib[AVR_SIBLEN-1] = 0; // Zero terminate string
```

Comment:
Well spotted!

Reply:
It took me quite a while to figure out why `PDATA(pgm)->set_sck()` did work when using `-B`, but not when communication was established. I wouldn't be surprised if I were the one who introduced the bug in the first place.